skbio/io/format/blast7.py

```diff
@@ -0,0 +1,387 @@
+==============================================
+
+.. currentmodule:: skbio.io.format.blast7
+
+The BLAST+7 format (``blast+7``) stores the results of a BLAST [1]_ database
+search. This format is produced by both BLAST+ output
+format 7 and legacy BLAST output format 9. The results
+are stored in a simple tabular format with headers. Values are separated by the
+tab character.
+
+An example BLAST+7-formatted file comparing two nucleotide sequences, taken
+from [2]_ (tab characters represented by ``<tab>``):
+
+.. code-block:: none
+
+    # BLASTN 2.2.18+
+    # Query: gi|1786181|gb|AE000111.1|AE000111
+    # Subject: ecoli
+    # Fields: query acc., subject acc., evalue, q. start, q. end, s. st\
+art, s. end
+    # 5 hits found
+    AE000111<tab>AE000111<tab>0.0<tab>1<tab>10596<tab>1<tab>10596
+    AE000111<tab>AE000174<tab>8e-30<tab>5565<tab>5671<tab>6928<tab>6821
+    AE000111<tab>AE000394<tab>1e-27<tab>5587<tab>5671<tab>135<tab>219
+    AE000111<tab>AE000425<tab>6e-26<tab>5587<tab>5671<tab>8552<tab>8468
+    AE000111<tab>AE000171<tab>3e-24<tab>5587<tab>5671<tab>2214<tab>2130
+
+Format Support
+==============
+**Has Sniffer: Yes**
+
+**State: Experimental as of 0.4.0-dev.**
+
++------+------+---------------------------------------------------------------+
+|Reader|Writer|                          Object Class                         |
++======+======+===============================================================+
+|Yes   |No    |:mod:`pandas.DataFrame`                                        |
++------+------+---------------------------------------------------------------+
+
+Format Specification
+====================
+There are two BLAST+7 file formats supported by scikit-bio: BLAST+ output
+format 7 (``-outfmt 7``) and legacy BLAST output format 9 (``-m 9``). Both file
+formats are structurally similar, with minor differences.
+
+Example BLAST+ output format 7 file::
+
+    # BLASTP 2.2.31+
+    # Query: query1
+    # Subject: subject2
+    # Fields: q. start, q. end, s. start, s. end, identical, mismatches, sbjct\
+frame, query acc.ver, subject acc.ver
+    # 2 hits found
+    1	8	3	10	8	0	1	query1	subject2
+    2	5	2	15	8	0	2	query1	subject2
+
+.. note:: Database searches without hits may occur in BLAST+ output format 7
+   files. scikit-bio ignores these "empty" records:
+
+   .. code-block:: none
+
+       # BLASTP 2.2.31+
+       # Query: query1
+       # Subject: subject1
+       # 0 hits found
+
+Example legacy BLAST output format 9 file:
+
+.. code-block:: none
+
+    # BLASTN 2.2.3 [May-13-2002]
+    # Database: other_vertebrate
+    # Query: AF178033
+    # Fields:
+    Query id,Subject id,% identity,alignment length,mismatches,gap openings,q.\
+ start,q. end,s. start,s. end,e-value,bit score
+    AF178033    EMORG:AF178033  100.00  811 0   0   1   811 1   811 0.0 1566.6
+    AF178033    EMORG:AF031394  99.63   811 3   0   1   811 99  909 0.0 1542.8
+
+.. note:: scikit-bio supports reading files that contain both these file
```

Comment:
Can you remove this note? Thinking about it more, this isn't something we should advertise about the reader, it's more of a side-effect of the implementation. Users probably shouldn't be concatenating BLAST+ and legacy BLAST files together.